### PR TITLE
Some small fixes / changes

### PR DESCRIPTION
**Pull request by madroach**
Feel free to cherry-pick whatever you like.

### CODE REVIEW

**Comment by Drup** [comment]
Good work! In general, the error handling is indeed very primitive so far. I think we should try to do more, but at least that's a good first step.

File: src/lmdb.ml [hunk 1]
```diff
@@ -159,7 +159,12 @@ let trivial_txn ~write env f =
...
   mdb_txn_begin env None txn_flag txn ;
-  let x = f !@txn in
+  let x =
+    try f !@txn with
+    |e ->
+      mdb_txn_abort !@txn ;
+      raise e
+  in
   mdb_txn_commit !@txn ;
   x
```
Comment (Drup) [inline src/lmdb.ml]: Actually, we might as well move the commit/return bits inside the try as well.

**Comment by madroach** [comment]
also note https://github.com/ocamllabs/ocaml-ctypes/issues/571

**Comment by Drup** [comment]
Wow, I appreciate all the work, but please calm down a little bit :D 

Can you split this whole thing into several PR: one for the fixes/direct API additions, one for the set of typing refinements for permissions, and one for the polymorphic API ? So that we can discuss them separately ?

**Comment by Drup** [comment]
(and honestly, since you seem very motivated to make lot's of chances, I wouldn't mind a patch to make CI/tests .. work). That would help me quite a lot in merging patches quickly.

**Comment by madroach** [comment]
I reset the PR to a branch just adding my benchmark resp. test, which should trigger the use-after-free bug I initially tried to fix.
What's wrong with CI/tests?

**Comment by madroach** [comment]
would you mind committing this PR now? I'll provide more PRs for my other changes.

**Comment by madroach** [comment]
Now also with the bugfix, so we get a nice travis result :-)

**Comment by Drup** [comment]
Great! Thanks.

**Pull request merged**